src/analysis/robust_decision_making.jl

```diff
@@ -0,0 +1,91 @@
+include("pareto.jl")
+
+"""
+    find_robust_clustering(result_set::ResultSet, metrics::Vector{ADRIA.metrics.Metric}; num_clusters::Int64=6)
+
+Find scenarios that perform well across a list of given metrics using clustering strategy:
+For each metric outcome:
+    - The scenarios are clusterized for each metric outcome.
+    - A median series is computed for each cluster.
+    - A summary statistics is computed for each median series (temporal variance).
+    -  Robust Clusters are those whose summary statistics is above the median.
+Robust Scenarios are, then, those that belong to the Robust Cluster for all metrics.
+
+# Arguments
+- `result_set` : ResultSet
+- `metrics` : Vector of metrics to consider when looking for robust clusters
+- `num_clusters` : Number of clusters to consider when clustering scenarios for each metric
+
+# Returns
+BitVector with true for robust scenarios and false for non-robust
+
+# Example
+```julia
+rs::ResultSet = ADRIA.run_scenarios(samples, domain, "45")
+
+_metrics::Vector{ADRIA.metrics.Metric} = [
+    ADRIA.metrics.scenario_total_cover,
+    ADRIA.metrics.scenario_asv,
+    ADRIA.metrics.scenario_absolute_juveniles,
+]
+
+robustness_limit::Float64 = 0.25
+
+robust_scens = ADRIA.analysis.find_robust_clustering(
+    rs, _metrics; robustness_limit=robustness_limit
+)
+```
+"""
+function find_robust_clustering(
+    result_set::ResultSet,
+    metrics::Vector{ADRIA.metrics.Metric};
+    num_clusters::Int64=6,
+    robustness_limit=0.2,
```

Comment:
This is essentially equal weighted search - I would prefer the user be able to pass in a function that defines what the robustness rule is, similar to the implementation for `outcome_map()`

https://github.com/open-AIMS/ADRIA.jl/blob/32c0a20ef86976ab344c7679c4ea995cf6471a9e/src/analysis/sensitivity.jl#L408